utils/rollout.js

```diff
@@ -0,0 +1,111 @@
+const crypto = require('crypto')
+
+const generateUUID = () => {
+  return crypto.randomBytes(8).toString('hex')
+}
+
+/**
+ * Calculate a numerical hash for a string.
+ * Refer:
+ *   - https://gist.github.com/hyamamoto/fd435505d29ebfa3d9716fd2be8d42f0
+ *   - https://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/
+ * @param {String} str: String for which hash number should be generated.
+ * @returns {Int}: A hash number for the input string.
+ */
+const hashCode = (str) => {
+  let hash = 0
+  for (let i = 0; i < str.length; i++) {
+    hash = Math.imul(31, hash) + str.charCodeAt(i) | 0
+  }
+  return Math.abs(hash)
+}
+
+/**
+ * Check if the user has any one of the roles that are allowed for this feature.
```

Comment:
Would it make sense to make both roles the same datatype?

Reply:
I feel like `array` is more suitable.

But the `user.roles` schema is a map in database (for reasons I cannot recall). 

It requires DB migration and changes in `authorization` middleware as well so I didn't bother with it.